Document advanced LWIP options (#4959)
OttoWinter/esphomedocs: components/esp32.rst

  the ESP-IDF framework. Not valid for the Arduino framework. Use of experimental features may cause instability or
  other issues.

**LWIP Optimization Options (ESP-IDF only):**

The following options are available under the ``advanced`` section when using the ESP-IDF framework to disable unused
LWIP (Lightweight IP) features and save flash memory (approximately 4KB):

- **enable_lwip_dhcp_server** (*Optional*, boolean): Enable DHCP server functionality. Only needed if the device will act

>   as a DHCP server (e.g., in WiFi AP mode). When the WiFi component is used, it automatically handles enabling/disabling

Reviewer: I've been trying to avoid "e.g." and "i.e." 🙂
```suggestion
  as a DHCP server (necessary for WiFi AP mode). When the WiFi component is used, it automatically handles enabling/disabling
```
